### PR TITLE
Week 8 Lab Notes

### PR DESCRIPTION
**Pull request by MrinalJain17**
In this section, we discussed a specific type of generative model called Variational Autoencoders and compared their functionalities and advantages over Classic Autoencoders. We explored the objective function of VAE in detail, understanding how it enforced some structure in the latent space. Finally, we implemented and trained a VAE on the MNIST dataset and used it to generate new samples.

### CODE REVIEW

File: docs/en/week08/08-3.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+date: 27 March 2020
+---
+
+## Introduction to Variational Autoencoder (VAE)
+
+### Recap: Auto-encoder (AE)
+
+To summarize at a high level, a very simple form of AE is as follows:
+
+- First, the autoencoder takes in an input and maps it to a hidden state through an affine transformation $\mathbf{h} = \sigma(\mathbf{W}_h \mathbf{x} + \mathbf{b}_h)$, where $\sigma$ is an (element-wise) activation function. This is the **encoder** stage. Note that $\mathbf{h}$ is also called the **code**.
```
Comment (Atcold) [inline docs/en/week08/08-3.md]: Use `\vect{}` or `\boldsymbol{}`, not `\mathbf{}`.

File: docs/en/week08/08-3.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+*In Figure 3 above, each bubble represents an estimated region of $z$, and the arrows represent how the reconstruction term pushes each estimated value away from the others, which is explained more below.*
+
+If there is overlap between any two estimates of $z$, (visually, if two bubbles overlap) this creates ambiguity for reconstruction because the points in the overlap can be mapped to both original inputs.  Therefore the reconstruction loss will push the points away from one another.
+
+However, if we use just the reconstruction loss, the estimates will continue to be pushed away from each other and the system could blow up.  This is where the penalty term comes in.
+
+**Note:** for binary inputs the reconstruction loss is
+
+$$
+l(x, \hat{x}) = - \sum\limits_{i=1}^n [x_i \text{log}(\hat{x_i}) + (1 - x_i)log(1-\hat{x_i})]
```
Comment (Atcold) [inline docs/en/week08/08-3.md]: Use `\log`!!!

File: docs/en/week08/08-3.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+<b>Fig. 4</b>: Plot showing how relative entropy forces the bubbles to have variance = 1
+</center>
+
+
+So we can see that this expression is minimized when $z_i$ has variance 1.  Therefore our penalty loss will keep the variance of our estimated latent variables at around 1.  Visually, this means our "bubbles" from above will have a radius of around 1.
+
+The last term, $\mathbb{E}(z_i)^2$, minimizes the distance between the $z_i$ and therefore prevents the "exploding" encouraged by the reconstruction term. 
+
+<center>
+<img src="{{site.baseurl}}/images/week08/08-3/fig_5.png" height="400px" width="500px"/><br>
```
Comment (Atcold) [inline docs/en/week08/08-3.md]: Remove `height` or `width`. Now the image has a bad aspect ratio!

File: docs/en/week08/08-3.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+<b>Fig. 5</b>: The "bubble-of-bubble" interpretation of VAE
+</center>
+
+*Figure 5 above shows how VAE loss pushed the estimated latent variables as close together as possible without any overlap while keeping the estimated variance of each point around one.*
+
+**Note:** The $\beta$ in the VAE loss function is a hyperparameter that dictates how to weight the reconstruction and penalty terms.
+
+---
```
Comment (Atcold) [inline docs/en/week08/08-3.md]: Remove.

File: docs/en/week08/08-3.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+$$
+l(x, \hat{x}) = \frac{1}{2} ||x - \hat{x} ||^2
+$$
+
+### The Penalty Term
+
+The second term is the relative entropy (a measure of the distance between two distributions) between $z$ which comes from a gaussian with mean $\mathbb{E}(z)$, variance $\mathbb{V}(z)$ and the standard normal distribution. If we expand this second term in the VAE loss function we get:
+
+$$
+\beta l_{\text{KL}}(z,\mathcal{N}(\textbf{0}, \mathbb{I}_d)) = \frac{\beta}{2} \sum\limits_{i=1}^d(\mathbb{V}(z) - \text{log}[\mathbb{V}(z)] - 1 + \mathbb{E}(z)^2)
```
Comment (Atcold) [inline docs/en/week08/08-3.md]: All `z` should be `z_i`.

File: docs/en/week08/08-3.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+
+
+## Breaking apart the VAE Loss Function
+
+### Visualizing Latent Variable Estimates and Reconstruction Loss
+
+As stated above, the loss function for the VAE contains two parts: a reconstruction term and a regularization term.  We can write this as
+
+$$
+l(x, \hat{x}) = l_{reconstruction} + \beta l_{\text{KL}}(z,\mathcal{N}(\textbf{0}, \mathbb{I}_d))
```
Comment (Atcold) [inline docs/en/week08/08-3.md]: `x` and `z` are vectors! They need a `\vect{}`!

**Comment by Atcold** [comment]
Pull `upstream`, add edits, send a new PR.

File: docs/en/week08/08-3.md (hunk 4 above)
Comment (MrinalJain17) [reply docs/en/week08/08-3.md]: Just to confirm, should I remove that note?

Comment (Atcold) [reply docs/en/week08/08-3.md]: The `---`.

**Pull request merged**